components/infobox/commons/infobox_person.lua

```diff
@@ -0,0 +1,289 @@
+
+local Language = mw.language.new('en')
+local _LINK_VARIANT = 'player'
+local shouldStoreData
+
+function Player.run(frame)
+	local player = Player(frame)
+	return player:createInfobox(frame)
+end
+
+function Player:createInfobox(frame)
+	local infobox = self.infobox
+	local args = self.args
+	shouldStoreData = Player:shouldStoreData(args)
+
+	local earnings = self:calculateEarnings(args)
+	Variables.varDefine('earnings', earnings)
+	if earnings == 0 then
+		earnings = nil
+	else
+		earnings = '$' .. Language:formatNum(earnings)
+	end
+	local nameDisplay = self:nameDisplay(args)
+	local role = self:getRole(args)
+	local birthDisplay, deathDisplay, birthday, deathday
+		= GetBirthAndDeath(
+			args.birth_date,
+			args.birth_location,
+			args.death_date,
+			role.category,
+			shouldStoreData
+		)
+	local status = self:getStatus(args)
+
+	infobox:name(nameDisplay)
+	infobox:image(args.image, args.defaultImage)
+	infobox:centeredCell(args.caption)
+	infobox:header(self:getInformationType(args) .. ' Information', true)
+	infobox:cell('Name', args.name)
+	infobox:cell('Romanized Name', args.romanized_name)
+	infobox:cell('Birth', birthDisplay)
+	infobox:cell('Died', deathDisplay)
+	infobox:cell('Status', status.display)
+	infobox:cell(role.title or 'Role', role.display)
+	infobox:fcell(Cell :new('Country')
+						:options({})
+						:content(
+							self:_createLocation(args.country or args.nationality, args.location, role.category),
+							self:_createLocation(args.country2 or args.nationality2, args.location2, role.category),
+							self:_createLocation(args.country3 or args.nationality3, args.location3, role.category)
+						)
+						:make()
+			)
+	infobox:fcell(Cell :new('Country')
+						:options({})
+						:content(unpack(self:_createLocations(args, role.category)))
+						:make()
+			)
+			:cell('Region', self:_createRegion(args.region))
+			:fcell(Cell :new('Team')
+						:options({})
+						:content(
+							self:_createTeam(args.team, args.teamlink),
+							self:_createTeam(args.team2, args.teamlink2)
+						)
+						:make()
+			)
+			:fcell(Cell :new('Clan')
+						:options({})
+						:content(
+							self:_createTeam(args.clan, args.clanlink),
+							self:_createTeam(args.clan2, args.clanlink2)
+						)
+						:make()
+			)
+	infobox:cell('Alternate IDs', args.ids or args.alternateids)
+	infobox:cell('Nicknames', args.nicknames)
+	infobox:cell('Total Earnings', earnings)
+	self:addCustomCells(infobox, args)
+
+	local links = Links.transform(args)
+	local achievements = self:getAchievements(infobox, args)
+	local history = self:getHistory(infobox, args)
+
+	infobox :header('Links', not Table.isEmpty(links))
+	infobox:links(links, _LINK_VARIANT)
+	infobox:header('Achievements', achievements)
+	infobox:centeredCell(achievements)
+	infobox:header('History', history)
+	infobox:centeredCell(history)
+	infobox:centeredCell(args.footnotes)
+	self:addCustomContent(infobox, args)
+	infobox:bottom(self:createBottomContent(infobox))
```

Comment:
Why does this function not have args available like the other overridable methods do?